src/array.jl

```diff
@@ -412,7 +415,7 @@ function copy!(dest::CategoricalArray{T, N}, dstart::Integer,
     if dstart == dstart == 1 && n == length(dest) == length(src)
```

Comment:
By adding `!isa(dest, SubArray) && ...` here, you could relax the method signature to accept `dest::SubArray`. I don't think any other changes are needed, but testing is of course needed to check that.